File: cs50/pset1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height, i, j;
    // Get user input for size of pyramid, looping until an appropriate input is given
    do
    {
        height = get_int("Height: \n");
    }
    while (height > 8 || height < 1);
    // Loops on loops to build the pyramid, one large cycle per row
    // Starts at the top
    for (j = 1; j < height + 1; j++)
    {
        // First section of empty space
        for (i = 0; i < height - j; i++)
        {
            printf(" ");
        }
        // Pre gap blocks
        for (i = height; i > height - j; i--)
        {
            printf("#");
        }
        // Make the gap, no loop reqd
        printf("  ");
        // print second half of blocks
        for (i = height; i > height - j; i--)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: cs50/pset6/mario.py
from cs50 import get_int
# infinite loop until an appropriate input is given
while True:
    # get height input from user
    height = get_int("Height: ")
    # check if input is appropriate
    if height > 0 and height < 9:
        # continue only after appropriate input is given
        break
    # repeatedly loop until everything is drawn
for i in range(height, 0, -1):
    for j in range(0, i - 1):
        print(" ", end = "")
    for j in range(height - i + 1):
        print("#", end = "")
    print("  ", end = "")
    for j in range(height - i + 1):
        print("#", end = "")
    print()